alert/tests.py

```diff
@@ -532,21 +534,341 @@ def test_no_permission(self):
+        self.assertEqual(response.data[0]['new_status'], 'O')
+        self.assertEqual(response.data[0]['alert_sent'], True)
+        self.assertFalse(hasattr(response.data[0], 'request_body'))
+
+    def test_cis121_missing(self):
+        response = self.client.get('/statusupdate/CIS-121/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(0, len(response.data))
+
 
-    def test_update_status(self):
-        self.section.status = 'C'
-        self.section.save()
-        up = record_update(self.section.full_code,
-                           TEST_SEMESTER,
-                           'C',
-                           'O',
-                           True,
-                           'JSON')
-        up.save()
-        update_course_from_record(up)
-        _, section = get_or_create_course_and_section(self.section.full_code, TEST_SEMESTER)
-        self.assertEqual('O', section.status)
+@override_settings(SWITCHBOARD_TEST_APP='pca')
+class UserDetailTestCase(TestCase):
+    def setUp(self):
+        User.objects.create_user(username='jacob',
+                                 password='top_secret')
+        self.client = APIClient()
+        self.client.login(username='jacob', password='top_secret')
+
+    def test_settings_before_create(self):
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual('jacob', response.data['username'])
+        self.assertEqual('', response.data['first_name'])
+        self.assertEqual('', response.data['last_name'])
+        self.assertEqual(None, response.data['profile']['email'])
+        self.assertEqual(None, response.data['profile']['phone'])
+
+    def test_update_settings(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob',
+                                               'profile': {'email': 'example@email.com', 'phone': '3131234567'}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_update_settings_change_first_name(self):
```

Comment:
Should add tests to make sure you can't change the settings of another user.

Reply:
Does `test_multiple_users_independent` cover that?